### PR TITLE
code refactor: zip unified, modules separated, direct commands usage removed, i18n

### LINKED ISSUE #10: Remove usage of general zip command
Remove usage of built-in `zip` command from the builder script to improve consistency and safety. Replace it with already used `best-zip` alternative.

### PR DESCRIPTION
**Pull request by bartoszglow**
Big code refactor for AEM Site Template Builder.

## Description

Refactor includes following improvements:
1. Fixes #10 
2. Removes usage of `zip` command, removes `bestzip` and replaces all with `adm-zip`.
3. Removes direct usage of `git` command (replaced by `simple-git`).
4. Provides catches for errors in internal scripts.
5. Modularized code.
6. Separated i18n.
7. Standardize way of logging info and errors.
8. ESLint integration.
9. Unit test coverage (100% lines, 98.28% stmts).

I propose to release version `2.1.0` of AEM Site Template Builder when these changes are merged in.

## Related Issue

#10 

## Motivation and Context

Changes make script more predictable, reliable and feature proof.

## How Has This Been Tested?

Changes tested with 4.0.4 version of [Basic Template](https://github.com/adobe/aem-site-template-basic) locally on cloud-ready quick start.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
 const PATHS = {
-  files: 'files',
-  previews: 'previews',
-  properties: 'package.json',
-  site: 'site',
-  theme: 'theme'
+  files: path.join(rootPath, 'files'),
+  packageJson: path.join(rootPath, 'package.json'),
+  previews: path.join(rootPath, 'previews'),
+  properties: path.join(rootPath, 'package.json'),
```
Comment (vladbailescu) [inline lib/config.js]: Isn't this a duplicate of L16?

**Review by comanV — APPROVED** [review]
the code looks good to me

File: .gitignore [hunk 2]
```diff
@@ -6,7 +6,13 @@ node_modules
 
 # resources created during test run
 .nyc_output
+test/resources/incorrect-site-template/incorrect-site-template-1.0.0.zip
+test/resources/incorrect-site-template/incorrect-site-template-latest.zip
+test/resources/incorrect-site-template/site-template
+test/resources/incorrect-site-template/site/target
+test/resources/incorrect-site-template/theme/package-lock.json
+test/resources/some-site-template/some-site-template-1.0.0.zip
```
Comment (vladbailescu) [inline .gitignore:14]: Can't we just add `test/resources/incorrect-site-template` and `test/resources/some-site-template` instead of all these details?

**Review by vladbailescu — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (bartoszglow) [reply .gitignore:14]: Generally we need to have the test site template in the repo so we can test it / against it. We want to omit only some folders that are getting generated automatically during the build (like the generated .zip files based on the test Site Template). The only thing we can do is to extract test related .gitignore entries into separate .gitignore file which would be placed in the test folder so it does not mix with the regular .gitignore rules.

**Pull request merged**